nre/bpf/nodefw/nodefw/src/main.rs

```diff
@@ -103,7 +107,36 @@ async fn main() -> Result<(), anyhow::Error> {
         listener,
         router,
     };
-    let _ = server::serve(sc).await;
+    loop {
+        tokio::select! {
+            biased;
+
+            _ = server::serve(sc) => {},
+            _ = return_on_killswitch(opt.killswitch_path) => {
+                warn!("firewall killswitch set, cancelling server...");
+                ctx.cancel();
+            },
+        }
+    }
     info!("firewall has stopped, exiting.");
     Ok(())
 }
+
+async fn return_on_killswitch(path: String) {
+    let path = std::path::Path::new(&path).join(KILLSWITCH_FILENAME);
+    loop {
+        match fs::read_to_string(path.clone()) {
+            Ok(state) => {
+                if state.trim().to_lowercase().as_str() == "on" {
+                    return;
+                }
+            }
+            Err(err) => {
+                if path.exists() {
+                    panic!("Failed to read firewall killswitch file: {}", err);
+                }
+            }
+        }
+        tokio::time::sleep(std::time::Duration::from_secs(3)).await;
+    }
+}
```

Comment:
can we refactor this a bit to be more like what is in https://github.com/MystenLabs/sui/pull/17195?

it's fine if you prefer to write text to the file, but the existence of the file is more than sufficient.  Please also utilize the cancellationtoken.